### PR TITLE
Add Rodentia

### PR DESCRIPTION
**Pull request by paige404**
## About the PR

Adds the Rodentia, a species of rat/mouse/squirrel/bat/hamster/rabbit/beaver/possum-like humanoids. As seen on [Delta-V](https://github.com/DeltaV-Station/Delta-v/pull/1511), [Goob](https://github.com/Goob-Station/Goob-Station/pull/610), [Cosmatic Drift](https://github.com/cosmatic-drift-14/cosmatic-drift/pull/404), and others.

Credit goes to @portfiend for initially developing the Rodentia for DeltaV.

## Why / Balance

We got dogs, cats. Why not rats? Many marines have been eager for this PR.

## Technical details

C#: Adds a new Last First naming scheme which Rodentia use elsewhere.
YAML: Adds rodentia, their associated sprites/markings, speech sounds, name conventions, etc.
PNG/JSON: Adds rodentia sprites, as well as custom mask sprites that fit the Rodentia Round Snout.

Note: the Bat Snout is too large and clips out of the rodentia gas mask by 1 pixel. This is because of #5173, which I may have a solution in the wings for. Expanding the gas mask even further looks awkward on default rodentia faces, I believe the best course of action will be to let the bat snout clip until the gas mask issue is resolved.

## Media

2024-12-16 22_29_33-Rouny's Marine Corps 14
2024-12-17 21_35_28-Rouny's Marine Corps 14
(The above screens show the dwarf-proportioned rodentia, which have since been made full size and are not accurate. They demonstrate a variety of outfits and equipment)

2024-12-17 22_38_18-Rouny's Marine Corps 14
(This screen shows a squirrel the size of a human man)

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- add: The Rodentia species are now enlisted in the marines.

### CODE REVIEW

File: Content.Shared/Humanoid/NamingSystem.cs [hunk 1]
```diff
@@ -35,6 +35,9 @@ public string GetName(string species, Gender? gender = null)
                 case SpeciesNaming.FirstDashFirst:
                     return Loc.GetString("namepreset-firstdashfirst",
                         ("first1", GetFirstName(speciesProto, gender)), ("first2", GetFirstName(speciesProto, gender)));
+                case SpeciesNaming.LastFirst:
+                    return Loc.GetString("namepreset-lastfirst",
+                        ("last", GetLastName(speciesProto)), ("first", GetFirstName(speciesProto, gender)));
```
Comment (QuietlyWhisper) [inline Content.Shared/Humanoid/NamingSystem.cs:40]: This should have a comment on it since it's an upstream file, so it won't be overwritten by an update

Something like "this is for [species]"

File: Content.Shared/Humanoid/Prototypes/SpeciesPrototype.cs [hunk 2]
```diff
@@ -128,4 +128,5 @@ public enum SpeciesNaming : byte
     FirstLast,
     FirstDashFirst,
     TheFirstofLast,
+    LastFirst,
```
Comment (QuietlyWhisper) [inline Content.Shared/Humanoid/Prototypes/SpeciesPrototype.cs:131]: This should have a comment on it since it's an upstream file, so it won't be overwritten by an update

Something like "this is for [species]"

File: Resources/Audio/_RMC14/Voice/Rodentia/mouse_squeak.ogg
Comment (QuietlyWhisper) [inline Resources/Audio/_RMC14/Voice/Rodentia/mouse_squeak.ogg:1]: This is a duplicate file of the mouse squeak used in mice mobs, you could probably reuse that one to save space.

File: Resources/Textures/_RMC14/Mobs/Rodentia/cheek_markings.rsi/meta.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "https://github.com/tgstation/tgstation/blob/8024397cc81c5f47f74cf4279e35728487d0a1a7/icons/mob/human_parts_greyscale.dmi, modified by DrSmugleaf and portfiend",
```
Comment (QuietlyWhisper) [inline Resources/Textures/_RMC14/Mobs/Rodentia/cheek_markings.rsi/meta.json:4]: Source is incorrect

File: Resources/Textures/_RMC14/Mobs/Rodentia/body_markings.rsi/meta.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "https://github.com/tgstation/tgstation/blob/8024397cc81c5f47f74cf4279e35728487d0a1a7/icons/mob/human_parts_greyscale.dmi, modified by DrSmugleaf and portfiend",
```
Comment (QuietlyWhisper) [inline Resources/Textures/_RMC14/Mobs/Rodentia/body_markings.rsi/meta.json:4]: Source is incorrect.

File: Resources/Audio/_RMC14/Voice/Rodentia/attributions.yml
Comment (QuietlyWhisper) [inline Resources/Audio/_RMC14/Voice/Rodentia/attributions.yml:1]: This is a duplicate file of the mouse squeak used in mice mobs, you could probably reuse that one to save space.

File: Resources/Textures/_RMC14/Mobs/Rodentia/head_markings.rsi/meta.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "https://github.com/tgstation/tgstation/blob/8024397cc81c5f47f74cf4279e35728487d0a1a7/icons/mob/human_parts_greyscale.dmi, modified by DrSmugleaf and portfiend",
```
Comment (QuietlyWhisper) [inline Resources/Textures/_RMC14/Mobs/Rodentia/head_markings.rsi/meta.json:4]: Source is incorrect.

**Review by QuietlyWhisper — CHANGES_REQUESTED** [review]
You have a bunch of sprites with no source linking to the tg body parts.

You'll need to find the source of these original sprites and link them, and double check the license used.

File: Resources/Audio/_RMC14/Voice/Rodentia/attributions.yml
Comment (paige404) [reply Resources/Audio/_RMC14/Voice/Rodentia/attributions.yml:1]: I could! I figured moving it might help if rodentia speech/emote sounds get updates. Which, I'd love to see those expanded both here and elsewhere in the future. I can reuse the animal mouse squeak for now if that's preferred though.

**Comment by paige404** [comment]
> You have a bunch of sprites with no source linking to the tg body parts.
> 
> You'll need to find the source of these original sprites and link them, and double check the license used.

That was the attribution on portfiend's sprites on DeltaV. I'll update these to be like the Vulpkanin's sprite attributions.

**Comment by paige404** [comment]
I have done my best to address all feedback. I noticed that while Delta-V's sprite metadata claims CC-BY-SA-3.0, their own code (and thus our usage of their sprites) is under AGPLv3. So I corrected the metadata for the imported sprites. Hopefully I've done it right this time?

**Comment by QuietlyWhisper** [comment]
> I have done my best to address all feedback. I noticed that while Delta-V's sprite metadata claims CC-BY-SA-3.0, their own code (and thus our usage of their sprites) is under AGPLv3.

The sprites used for vulp are from a commit before deltav was AGPLv3, they're also not ported directly from deltav but from another server that already handled the problem, which you'll essentially need to do here as well, unless they were added during/after AGPLv3, in which case we can't accept these sprites / whatever was sourced from deltav under any circumstances due to the license conflict. Sorry that you had put so much effort into this, but it's very unlikely we'll be able to add these for reasons out of control of either of us.

**Comment by paige404** [comment]
> > I have done my best to address all feedback. I noticed that while Delta-V's sprite metadata claims CC-BY-SA-3.0, their own code (and thus our usage of their sprites) is under AGPLv3.
> 
> The sprites used for vulp are from a commit before deltav was AGPLv3, they're also not ported directly from deltav but from another server that already handled the problem, which you'll essentially need to do here as well, unless they were added during/after AGPLv3, in which case we can't accept these sprites / whatever was sourced from deltav under any circumstances due to the license conflict.

The vulp attribution lines specifically [do say](https://github.com/RMC-14/RMC-14/pull/4699/files#diff-42da705d201c8454e5eff0fe2133bfb9a25b30449f993269f3014eff16b00915R4) that they are ported from Delta-V.

Rodentia were added after the code went AGPL. _However_, DeltaV's listed license for these sprites is CC BY SA 3.0 as you can see on their own meta.json files (e.g. [this one](https://github.com/DeltaV-Station/Delta-v/blob/f49142ca8d7ceac483094444c372e849309dc02b/Resources/Textures/DeltaV/Mobs/Customization/Rodentia/body_markings.rsi/meta.json)). Additionally, the creator of the PR gave consent for ports in [the PR itself](https://github.com/DeltaV-Station/Delta-v/pull/1511).

**Comment by deltanedas** [comment]
AGPL is for code, assets like sprites (and *especially* ones that are -SA) are still the same license

**Comment by QuietlyWhisper** [comment]
> The vulp attribution lines specifically [do say](https://github.com/RMC-14/RMC-14/pull/4699/files#diff-42da705d201c8454e5eff0fe2133bfb9a25b30449f993269f3014eff16b00915R4) that they are ported from Delta-V.

Yes, because it was explicitly copied from the fork that provided it, the sprites come from deltav, not the fork, obviously.

**Comment by QuietlyWhisper** [comment]
> Additionally, the creator of the PR gave consent for ports in [the PR itself](https://github.com/DeltaV-Station/Delta-v/pull/1511#issuecomment-2339581002).

Thanks for sharing this supporting information, it should be enough to be perfectly fine with this, I'll run it by smug as well and ask a few others just in case.

It's probably safe to assume the creator doesn't really understand how licensing works but it should be fine to port now.

image

**Review by QuietlyWhisper — APPROVED** [review]
You could probably add more xenonid translations for "rodent" and "rodents" if you care, but otherwise, assuming everything here has been tested to work and render correctly, this seems good to go. I'd still like smug to do their own review regardless as usual.

Additionally, you're absolutely invited to make a wiki addition to the species page, as the person advocating for this species.
https://wiki.rouny-ss14.com/wiki/Species

**Comment by paige404** [comment]
> You could probably add more xenonid translations for "rodent" and "rodents" if you care, but otherwise, assuming everything here has been tested to work and render correctly, this seems good to go. I'd still like smug to do their own review regardless as usual.
> 
> Additionally, you're absolutely invited to make a wiki addition to the species page, as the person advocating for this species. https://wiki.rouny-ss14.com/wiki/Species

At this point, I would rather do that chatsan expansion as a tiny follow-up PR. Probably after I've finished working on next PR, which is to fix issue #5173.

I'll gladly add Rodentia to the wiki though!

**Comment by paige404** [comment]
Oh! And regarding testing: I've been pretty thorough. I tested every armor I could find, most of the CM outfits available through the Debug > Set Outfit command, tested all the markings, tested gas mask, coif, rebreather, and sterile mask (it seems like the medical gas mask isn't really used? if that changes I'm happy to work out a sprite for that too). Tested xeno interactions like parasite latching, devouring, wall nesting, chestbursting. Tested carrying, dragging, wielding weapons. I can't guarantee there aren't any edge cases I missed, but I've been as thorough as possible.

I didn't update all my PR screenshots for the full-sized rats, but I've continued to test quite a lot since sizing them up.

**Comment by paige404** [comment]
> Additionally, you're absolutely invited to make a wiki addition to the species page, as the person advocating for this species. https://wiki.rouny-ss14.com/wiki/Species

How's this? https://wiki.rouny-ss14.com/wiki/Species

**Review by DrSmugleaf — APPROVED** [review]

**Comment by DrSmugleaf** [comment]
Thanks for the PR!
Shoutout to the people reacting to this and other species PRs with a thumbs down as if we give a fuck when reviewing or merging PRs!

**Comment by deltanedas** [comment]
nooooooo you cant port species time to out myself with a react
